### PR TITLE
Features/vfx

### PR DESCRIPTION
**Pull request by raulgonzalezupc**
Implementation of lightBall and some combat effects

### CODE REVIEW

File: Scripts/src/Script/LightBall.cpp [hunk 1]
```diff
@@ -52,6 +55,36 @@ void LightBall::Awake()
...
+		}
+		if (go->name == "blue_sparkle")
+		{
+			blue_sparkle = static_cast<ComponentParticleSystem*>(go->GetComponent(Component::ComponentType::PARTICLE_SYSTEM));
+		}
+		if (go->name == "purple_sparkle")
+		{
+			purple_sparkle = static_cast<ComponentParticleSystem*>(go->GetComponent(Component::ComponentType::PARTICLE_SYSTEM));
+		}
+	}
```
Comment (JorxSS) [inline Scripts/src/Script/LightBall.cpp:87]: Encapsulate this into a function could be better

File: Scripts/src/Script/LightBall.cpp [hunk 2]
```diff
@@ -111,6 +144,15 @@ void LightBall::Update()
 
 			owner->transform.SetGlobalMatrixTranslation(current_player->transform.GetGlobalTranslation() + float3::unitY 
 				+ current_player->transform.GetFrontVector() * cast_distance);
+
+			light_LB->active = false;
+			interior_texture_LB->active = false;
+			exterior_texture_LB->active = false;
+			small_texture_LB->active = false;
+			blue_sparkle->active = false;
+			purple_sparkle->active = false;
```
Comment (JorxSS) [inline Scripts/src/Script/LightBall.cpp:153]: Same as before, something like` SetLightballEffects (bool state)`

File: Scripts/src/Script/LightBall.cpp [hunk 3]
```diff
@@ -122,22 +164,57 @@ void LightBall::Update()
 				ball_enabled = false;
 				owner->transform.SetScale(cast_scale);
 				position_while_casting = 0.f;
+				interior_texture_LB->width = width_interior_texture_LB; interior_texture_LB->height = height_interior_texture_LB;
+				interior_texture_LB->width *= 0.5; interior_texture_LB->height *= 0.5;
+				exterior_texture_LB->width = width_exterior_texture_LB; exterior_texture_LB->height = height_exterior_texture_LB;
+				exterior_texture_LB->width *= 0.5; exterior_texture_LB->height *= 0.5;
+				small_texture_LB->width = width_small_texture_LB; small_texture_LB->height = height_small_texture_LB;
+				small_texture_LB->width *= 0.5; small_texture_LB->height *= 0.5;
```
Comment (JorxSS) [inline Scripts/src/Script/LightBall.cpp:172]: same as before example `SetSizeEffects(const float width, const float height) `

File: Scripts/src/Script/LightBall.cpp [hunk 4]
```diff
@@ -122,22 +164,57 @@ void LightBall::Update()
...
 					owner->children[0]->SetEnabled(true);
+					owner->children[0]->GetComponent(Component::ComponentType::MESH_RENDERER)->active = false;
+					light_LB->active = true;
+					interior_texture_LB->active = true; interior_texture_LB->loop = true; interior_texture_LB->Play(); 
+					interior_texture_LB->color[0] = 0.12157F; interior_texture_LB->color[1] = 0.68235F; interior_texture_LB->color[3] = 1.00000F;
+					exterior_texture_LB->active = true; exterior_texture_LB->loop = true; exterior_texture_LB->Play();
+					exterior_texture_LB->color[0] = 0.71373F; exterior_texture_LB->color[1] = 0.19216F; exterior_texture_LB->color[3] = 1.00000F;
+					small_texture_LB->active = true; small_texture_LB->loop = true; small_texture_LB->Play();
+					small_texture_LB->color[0] = 0.73725F; small_texture_LB->color[1] = 0.57255F; small_texture_LB->color[3] = 1.00000F;
+
```
Comment (JorxSS) [inline Scripts/src/Script/LightBall.cpp:188]: same as before `InitBallEffects()`

File: Scripts/src/Script/EnemyController.cpp [hunk 5]
```diff
@@ -306,6 +319,13 @@ void EnemyController::HitVFX()
 		hit_effect->Play();
 	}
 }
+void EnemyController::HitLightBallVFX()
+{
+	LB_hit_effect_purple->Emit(2);
+	LB_hit_effect_emerald->Emit(1);
+	LB_hit_effect_blue->Emit(4);
+
+}
```
Comment (Dartemiss) [inline Scripts/src/Script/EnemyController.cpp:328]: I think this can be const

File: Scripts/src/Script/LightBall.h [hunk 6]
```diff
@@ -64,7 +67,16 @@ class LightBall : public Script
 	EnemyManager* enemy_manager = nullptr; 
 	
 	ComponentAudioSource* audio_source = nullptr;
+	ComponentBillboard* interior_texture_LB = nullptr;
+	float width_interior_texture_LB, height_interior_texture_LB;
+	ComponentBillboard* exterior_texture_LB = nullptr;
+	float width_exterior_texture_LB, height_exterior_texture_LB;
+	ComponentBillboard* small_texture_LB = nullptr;
+	float width_small_texture_LB, height_small_texture_LB;
```
Comment (JorxSS) [inline Scripts/src/Script/LightBall.h:75]: init floats

File: Scripts/src/Script/LightOrb.cpp [hunk 7]
```diff
@@ -119,6 +119,7 @@ void LightOrb::Update()
 			player_to_delvier_controller->Heal(healing_value);
 			player_to_delvier_controller->FillSpecialBar(special_value);
 			audio_source->PlayEvent("play_orbs_pickup");
+			player_to_delvier_controller->EmitHealVFX();
```
Comment (JorxSS) [inline Scripts/src/Script/LightOrb.cpp:122]: small thing, `player_to_deliver`

File: Scripts/src/Script/LightBall.cpp (hunk 1 above)
Comment (raulgonzalezupc) [reply Scripts/src/Script/LightBall.cpp:87]: done!

File: Scripts/src/Script/LightBall.cpp (hunk 2 above)
Comment (raulgonzalezupc) [reply Scripts/src/Script/LightBall.cpp:153]: done!

File: Scripts/src/Script/PlayerController.cpp [hunk 8]
```diff
@@ -5,6 +5,7 @@
 #include "Component/ComponentBillboard.h"
 #include "Component/ComponentCollider.h"
 #include "Component/ComponentScript.h"
+#include "Component/ComponentParticleSystem.h"
```
Comment (JorxSS) [inline Scripts/src/Script/PlayerController.cpp:8]: CS alphabetical order :imp:

File: Scripts/src/Script/PlayerController.cpp [hunk 9]
```diff
@@ -261,6 +282,10 @@ void PlayerController::OnInspector(ImGuiContext* context)
 
 	ImGui::Spacing();
 
+	ImGui::DragFloat("Healt Time", &heal_time);
```
Comment (JorxSS) [inline Scripts/src/Script/PlayerController.cpp:285]: Heal without t

File: Scripts/src/Script/PlayerController.cpp [hunk 10]
```diff
@@ -421,4 +446,14 @@ void PlayerController::Dash()
 	is_dashing = true;
 	animation->ActiveAnimation("dash");
 	audio_source->PlayEvent("play_dash_player");
+}
+
+void PlayerController::EmitFromImpactPlayerBallMushdoom()
+{
+	splash_musdhoom_range_VFX->Emit(6);
+}
```
Comment (JorxSS) [inline Scripts/src/Script/PlayerController.cpp:454]: make this const

File: Scripts/src/Script/PlayerController.cpp [hunk 11]
```diff
@@ -421,4 +446,14 @@ void PlayerController::Dash()
...
+void PlayerController::EmitHealVFX()
+{
+	heal_VFX->Play();	
+	heal_vfx_active = true;
```
Comment (Dartemiss) [inline Scripts/src/Script/PlayerController.cpp:458]: Const

**Pull request merged**